config/nat.py

```diff
@@ -0,0 +1,1127 @@
+        return twice_id_count
+
+    for key,values in static_dict.items():
+        twice_id = 0
+
+        if "twice_nat_id" in values:
+            twice_id = int(values["twice_nat_id"])
+        else:
+            continue
+
+        if twice_id == twice_nat_id:
+            twice_id_count += 1
+
+    return twice_id_count
+
+def getTwiceNatIdCountWithDynamicBinding(twice_nat_id, count, dynamic_key):
+    """Get the twice nat id count with dynamic binding"""
+
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    nat_binding_dict = config_db.get_table('NAT_BINDINGS')
+    twice_id_count = count
+
+    if not nat_binding_dict:
+        return twice_id_count
+
+    for key, values in nat_binding_dict.items():
+        nat_pool_data = config_db.get_entry('NAT_POOL',values["nat_pool"])
+        twice_id = 0
+
+        if dynamic_key is not None:
+            if dynamic_key == key:
+                continue
+
+        if not nat_pool_data:
+            continue
+
+        if "twice_nat_id" in values:
+            if values["twice_nat_id"] == "NULL":
+                continue
+            else:
+                twice_id = int(values["twice_nat_id"])
+        else:
+            continue
+
+        if twice_id == twice_nat_id:
+            twice_id_count += 1
+
+    return twice_id_count
+ 
+############### NAT Configuration ##################
+
+#
+# 'nat' group ('config nat ...')
+#
+@click.group('nat')
+def nat():
+    """NAT-related configuration tasks"""
+    pass
+
+#
+# 'nat add' group ('config nat add ...')
+#
+@nat.group('add')
+def add():
+    """Add NAT-related configutation tasks"""
+    pass
+
+#
+# 'nat remove' group ('config nat remove ...')
+#
+@nat.group('remove')
+def remove():
+    """Remove NAT-related configutation tasks"""
+    pass
+
+#
+# 'nat set' group ('config nat set ...')
+#
+@nat.group('set')
+def set():
+    """Set NAT-related timeout configutation tasks"""
+    pass
+
+#
+# 'nat reset' group ('config nat reset ...')
+#
+@nat.group('reset')
+def reset():
+    """Reset NAT-related timeout configutation tasks"""
+    pass
+
+#
+# 'nat add static' group ('config nat add static ...')
+#
+@add.group('static')
+def static():
+    """Add Static related configutation"""
+    pass
+
+#
+# 'nat add static basic' command ('config nat add static basic <global-ip> <local-ip>')
+#
+@static.command('basic')
+@click.pass_context
+@click.argument('global_ip', metavar='<global_ip>', required=True)
+@click.argument('local_ip', metavar='<local_ip>', required=True)
+@click.option('-nat_type', metavar='<nat_type>', required=False, type=click.Choice(["snat", "dnat"]), help="Set nat type")
+@click.option('-twice_nat_id', metavar='<twice_nat_id>', required=False, type=click.IntRange(1, 9999), help="Set the twice nat id")
+def add_basic(ctx, global_ip, local_ip, nat_type, twice_nat_id):
+    """Add Static NAT-related configutation"""
+
+    # Verify the ip address format 
+    if is_valid_ipv4_address(local_ip) is False:
```

Comment:
The function is_valid_ipv4_address rejects if the ip is reserved IP address. Will this block use of some private ip ranges as local ip  address ?

Reply:
@arlakshm No, It won't block the private ip addresses. Function "is_reserved()" will check for IETF reserved ip addresses (Ex. 240.0.0.0/4). For private address checking, "is_private()" function is used.